css/ast/src/token.rs

```diff
@@ -17,6 +17,8 @@ pub struct TokenAndSpan {
 
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub enum Token {
+    EOF,
```

Comment:
Can other token come after `EOF` token?
I'm not sure how can it help error recovery.

Reply:
@kdy1 

> Can other token come after EOF token?

No, EOF is last

> I'm not sure how can it help error recovery.

There is an interesting situation here - I saw chrome CSS lexer and Servo CSS lexer (other implementations too), they all use it because it is clearly described in the specification (even `esbuild` uses it). 

One more problem - we use `ErrorKind::Eof` to indicate end of file, but actually there is no error here, we have to rely on token flow. This makes our code cleaner according to the spec.

In fact - yes, this does not fundamentally change nothing, but to be honest, I would like to be closer to the specification and not add anything new to it.

Here `esbuild` example of `EOF` token - https://github.com/evanw/esbuild/blob/master/internal/css_lexer/css_lexer.go#L257, if you need I can provide more links on sources

Reply:
Interesting. I don't like adding `Invalid` state to ast, (another token after `EOF` in this case), but I think there should be enough reason, so I'll merge it. Thanks!